### ISSUE TITLE
Incoming payments QUEUE and FAIL

### ISSUE THREAD

**Issue by strnly** [Blockstream/greenlight#461]
Had dozens of incoming payments failing. 
Outgoing payments work fine.

Example below

<img width="464" alt="Screenshot 2024-06-05 at 5 01 06 PM" src="https://github.com/Blockstream/greenlight/assets/25166070/fda3b743-e568-4874-bba4-6b1e9970b864">
<img width="461" alt="Screenshot 2024-06-05 at 5 01 27 PM" src="https://github.com/Blockstream/greenlight/assets/25166070/3d1eb776-4b00-424c-b58e-de2791402e95">

Invoice 

lnbc411670n1pnxqumksp50nhlajjkuvwq4zf3tzrw0n95945sagjjv6pwklld3w4f7vp3hx8spp52a6800f05way3uggyykyvtvsqayd5ev63a7zuvur99hfnr8jxwasdpl2pskjepqw3hjq5652f8yckfq9p8hyer9wgsyj3p6yqck6argtac8yetdd96k62gxqzurcqzysrzjqfhz36cfs6r34246yymzw6x65gfymn2azc2wnqg4txn0j854yddjz286f35sl4e6uqqqqqlgqqqq86qqyg9qxpqysgqjg00r7mugqqq8g82y0u82w0dmmhp29evcj0ujyqta0g870jhyc0j3t48yjdltmrkhzgj594ppglw46mhxqe52smxhdlgzsx74hw6csspwe755r

**Comment by cdecker** [comment]
I fetched the logs for the last 250 runs, looking for `WIRE_UPDATE_ADD_HTLC` which is the message used to add an HTLC to the channel. It'll surface both incoming and outgoing HTLCs, and if there are no matching HTLCs being added to the payment, then the payment must've gotten stuck before reaching GL.

I found the following instances of `WIRE_UPDATE_ADD_HTLC`:

 - `session_id=7203393256816640`
   - 2024-06-03T16:08:11+02:00 (`peer_in` HTLC 22) this is the LSP extending an HTLC to the node
   - 2024-06-03T16:08:12+02:00 (`peer_in` HTLC 23) another HTLC in the MPP from above
   - 2024-06-03T16:08:12+02:00 (`peer_in` HTLC 24) same here
   - 2024-06-03T16:08:14+02:00 HTLC set contains 3 HTLCs, for a total of 124307750msat out of 142066000msat (payment_secret)
   - 2024-06-03T16:09:23+02:00 `FAIL:: HTLC REMOTE 22 = SENT_REMOVE_HTLC/RCVD_REMOVE_HTLC`
   - 2024-06-03T16:09:23+02:00 `FAIL:: HTLC REMOTE 23 = SENT_REMOVE_HTLC/RCVD_REMOVE_HTLC`
   - 2024-06-03T16:09:23+02:00 `FAIL:: HTLC REMOTE 24 = SENT_REMOVE_HTLC/RCVD_REMOVE_HTLC`
   - As you can see the node receives a partial payment of 124307750msat out of a total of 142066000msat, waits for the remainder to come in, and then returns the partial payment with `MPP_TIMEOUT` after 60 seconds of waiting.
  - `session_id=7203406552760322`
    - 2024-06-03T16:50:42+02:00 (`peer_in` HTLC 25) 
    - 2024-06-03T16:50:44+02:00 `payment failed: Unknown invoice 9a0888d8c4b533f4b78b3f92f0c591950976ed3969dc8a09dca4efb6ea73579b`
    - The payment appears to not match an invoice we have internally. NOtice that this could also be an expired invoice, since those are treated the same way as non-existent invoices (prevent scanning for recipients of payments).
    - 2024-06-03T16:50:47+02:00 (`peer_in` HTLC 26) another payment?
    - 2024-06-03T16:50:49+02:00 `HTLC set contains 1 HTLCs, for a total of 5263000msat out of 5263000msat (payment_secret)`
    - 2024-06-03T16:50:49+02:00 Resolved invoice '{\"unix_milli\":1717426205654,\"payer_amount_msat\":null}' with amount 5263000msat in 1 htlcs
  - `session_id=7204140690178048`
    - 2024-06-05T17:35:33+02:00 (`peer_out` HTLC 476) sending a single-part payment
    - 2024-06-05T17:36:47+02:00 (`peer_in`) `FULFILL:: HTLC LOCAL 476 = RCVD_REMOVE_HTLC/SENT_REMOVE_HTLC FULFILLED`

So as far as we can tell from these logs, there have been 2 attempted incoming payments on Monday 3rd. The first one failed, because it only arrived partially, so the sender must've given up at some point. The cause could be in the network, on the sender or on the LSP, but not on GL, because we never saw that last part. The second payment went through correctly. And the last isntance where we saw HTLCs was an outgoing payment, that also succeeded.


The node is rather impressive by the way, because it only takes a RTT=80ms to send the signer request, and get a response.

**Comment by strnly** [comment]
Sender is me. I am using WoS on a 5G connected phone. Payments to my own node from the exact same device and app work flawlessly.

**Comment by cdecker** [comment]
Yes, there is no contradiction here, it appears that WoS is unable to route through the network to the nodes LSP, or the LSP is failing to forward the payment. Notice that the node has to be running when you use WoS to send to it (the node wallet had to be the front app on mobile for example). The node never seeing the HTLCs means that the payment fails in network rather than due to something we can control.
